### ISSUE TITLE
Memory footprint seems *way* too big

### ISSUE THREAD

**Issue by Floppy** [manyfold3d/manyfold#1784]
On try.manyfold.app, we can see some metrics for the app, and it's using around 500M of memory straight after boot, which slowly increases over time. This seems *way* too much, it shouldn't need to be like that.

I've removed some Rails components in #1782, and improved memory use while file streaming with #1776, but there's more to do.

**Comment by Floppy** [comment]
Tried #2987, which didn't make a huge difference because of musl probably. More to try in configuration here:

* https://www.joyfulbikeshedding.com/blog/2019-03-14-what-causes-ruby-memory-bloat.html#a-magic-trick-trimming
* https://matthaliski.com/blog/upgrading-to-rails-7-1-ruby-3-3-and-jemalloc
* https://github.com/mastodon/mastodon/blob/1fc165de02d79294c8a218f5fa82bcd477484ca1/Dockerfile#L67
* https://jemalloc.net/jemalloc.3.html#tuning

**Comment by Floppy** [comment]
after jemalloc, It's still high usage, but it does seem to be holding steadyish rather than growing continuously.

**Comment by tommyalatalo** [comment]
I can add to this issue; I just observed a constant increase of memory consumption by doing this:
1. Open a collection (I have one with currently 13 models) from the `Collections` view
2. See the memory usage increase by about 2-5%
3. Click the `Collections` link at the top

By repeating steps 1-3, i.e. opening the same collection over and over again I can increase the app memory usage until my computer locks up due to memory exhaustion. There is definitely a memory leak here where the assets are not released from memory when going back to the `Collections` view.

**Comment by Floppy** [comment]
That's really useful thank you; memory leaks are always hard to find, and replication is super useful. Do you get the same thing loading the models page? Collections should be *simpler*....!

**Comment by tommyalatalo** [comment]
Yes, the memory leak is the same if I just jump between `Models` and `Creators` in the top menu, loading the `Models` page keeps accumulating memory usage and clearly not releasing memory as it should when navigating to the `Creators` page. When I close the tab running manyfold I see all the memory getting released.

**Comment by Floppy** [comment]
OK, that makes sense at least. There's something hogging and not releasing memory in the renderer setup I think, which also explains why some pages just straight run out of memory and crash, if they have a lot of models, even before the rendering proper starts. (#4455)

**Comment by coderofsalvation** [comment]
I'm also hitting suspicious memory increase..

When I start the docker, everything idles around 100MB until PUMA/Jobs kick in to scan the files..then it'll idle between ~800M-1G.
